src/components/input-time/input-time.tsx

```diff
@@ -0,0 +1,872 @@
+   */
+  @Prop() readonly = false;
+
+  /**
+   * When true, makes the component required for form-submission.
+   *
+   * @internal
+   */
+  @Prop({ reflect: true }) required = false;
+
+  /** The scale (size) of the time input */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /** number (seconds) that specifies the granularity that the value must adhere to */
+  @Prop() step = 60;
+
+  /** The selected time in UTC (always 24-hour format) */
+  @Prop({ mutable: true }) value: string = null;
+
+  @Watch("value")
+  valueWatcher(newValue: string): void {
+    if (!this.userChangedValue) {
+      this.setValue({ newValue, context: "direct" });
+    }
+    this.userChangedValue = false;
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  private activeEl: HTMLSpanElement;
+
+  defaultValue: InputTime["value"];
+
+  formEl: HTMLFormElement;
+
+  private hourEl: HTMLSpanElement;
+
+  labelEl: HTMLCalciteLabelElement;
+
+  private meridiemEl: HTMLSpanElement;
+
+  private meridiemOrder: number;
+
+  private minuteEl: HTMLSpanElement;
+
+  private previousEmittedValue: string;
+
+  previousValue: string;
+
+  private secondEl: HTMLSpanElement;
+
+  private userChangedValue = false;
+
+  // --------------------------------------------------------------------------
+  //
+  //  State
+  //
+  // --------------------------------------------------------------------------
+
+  @State() hour: string;
+
+  @State() hourCycle: HourCycle;
+
+  @State() localizedHour: string;
+
+  @State() localizedHourSuffix: string;
+
+  @State() localizedMeridiem: string;
+
+  @State() localizedMinute: string;
+
+  @State() localizedMinuteSuffix: string;
+
+  @State() localizedSecond: string;
+
+  @State() localizedSecondSuffix: string;
+
+  @State() meridiem: Meridiem;
+
+  @State() minute: string;
+
+  @State() second: string;
+
+  @State() showSecond: boolean = this.step < 60;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Fires when the time value is committed.
+   */
+  @Event() calciteInputTimeChange: EventEmitter<string>;
+
+  /**
+   * Fires each time the user changes the value but has not committed changes.
+   */
+  @Event() calciteInputTimeInput: EventEmitter<string>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInternalInputTimeBlur: EventEmitter<void>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInternalInputTimeFocus: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("blur")
```

Comment:
we need to create some documentation alignment on how we are handling event listeners. I prefer this syntax but other devs are saying these should be setup as on event listeners in the component render itself. @benelan thoughts?

Reply:
Yeah I don't feel strongly in either direction but I agree on the need for consistency. I'll run it by Franco when he returns.